src/main/java/com/bisnode/opa/client/OpaClient.java

```diff
@@ -0,0 +1,48 @@
+package com.bisnode.opa.client;
+
+
+import com.bisnode.opa.client.data.OpaDataApi;
+import com.bisnode.opa.client.data.OpaDataClient;
+import com.bisnode.opa.client.policy.OpaPolicyApi;
+import com.bisnode.opa.client.policy.OpaPolicyClient;
+import com.bisnode.opa.client.query.OpaQueryApi;
+import com.bisnode.opa.client.query.OpaQueryClient;
+import com.bisnode.opa.client.rest.OpaRestClient;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Delegate;
+import lombok.experimental.FieldDefaults;
+
+import java.net.http.HttpClient;
+
+import static lombok.AccessLevel.PRIVATE;
+
+@FieldDefaults(makeFinal = true, level = PRIVATE)
+@RequiredArgsConstructor(access = PRIVATE)
+public class OpaClient implements OpaQueryApi, OpaDataApi, OpaPolicyApi {
+
+    @Delegate
+    OpaQueryApi opaQueryApi;
+    @Delegate
+    OpaDataApi opaDataApi;
+    @Delegate
+    OpaPolicyApi opaPolicyApi;
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        OpaConfiguration opaConfiguration;
+
+        public Builder opaConfiguration(String host) {
```

Comment:
I think "url" would be a better name here, given how the protocol and port is also (possibly) included.